### PR TITLE
Better error handling for malformed urls

### LINKED ISSUE #2394: URL parse needs better error
a weird stacktrace popped into my log out of nowhere and i have no idea waht triggered it or what it means

paste of the code cause githubs "code" removes formatting https://pastebin.com/uFgw9hyk

### PR DESCRIPTION
**Pull request by sjsadowski**
Closes #2394

### CODE REVIEW

**Comment by Tronic** [comment]
@sjsadowski Can you add tests for 100 % patch coverage?

File: sanic/request.py [hunk 1]
```diff
@@ -130,7 +131,12 @@ def __init__(
 
         self.raw_url = url_bytes
         # TODO: Content-Encoding detection
-        self._parsed_url = parse_url(url_bytes)
+        try:
+            self._parsed_url = parse_url(url_bytes)
+        except HttpParserInvalidURLError as InvalidURLError:
+            raise InvalidUsage(
+                f"URL is invalid or malformed"
```
Comment (prryplatypus) [inline sanic/request.py:138]:
```suggestion
                "URL is invalid or malformed"
```

**Comment by sjsadowski** [comment]
> @sjsadowski Can you add tests for 100 % patch coverage?

There was a test added for this in test_bad_request.py

**Review by ahopkins — CHANGES_REQUESTED** [review]
Looks like this branches off #2389 so this PR is blocked.

**Comment by sjsadowski** [comment]
@ahopkins my bad. I can reimplement off main.

**Comment by ahopkins** [comment]
> @ahopkins my bad. I can reimplement off main.

no need. We'll just do them in order.

**Comment by sjsadowski** [comment]
Closing because I need to rethink #2389 on which this is dependent.

**Pull request closed without merging**